### PR TITLE
Restore ProtocolLib 4.x series support

### PR DESCRIPTION
**Pull request by GatitoUwU**
Fix for 1.8.8 when using ProtocolLib 4.8.0 :D
```
> [14:02:59 ERROR]: [BungeeGuard] Unhandled exception occured in onPacketReceiving(PacketEvent) for BungeeGuard
java.lang.NoClassDefFoundError: com/comphenix/protocol/injector/temporary/TemporaryPlayerFactory
at me.lucko.bungeeguard.spigot.listener.ProtocolHandshakeListener.closeConnection(ProtocolHandshakeListener.java:134) ~[BungeeGuard.jar:?]
at me.lucko.bungeeguard.spigot.listener.ProtocolHandshakeListener.access$300(ProtocolHandshakeListener.java:55) ~[BungeeGuard.jar:?]
at me.lucko.bungeeguard.spigot.listener.ProtocolHandshakeListener$Adapter.onPacketReceiving(ProtocolHandshakeListener.java:104) ~[BungeeGuard.jar:?]
at com.comphenix.protocol.injector.SortedPacketListenerList.invokeReceivingListener(SortedPacketListenerList.java:114) [ProtocolLib-4.5.1.jar:4.5.1]
at com.comphenix.protocol.injector.SortedPacketListenerList.invokePacketRecieving(SortedPacketListenerList.java:67) [ProtocolLib-4.5.1.jar:4.5.1]
at com.comphenix.protocol.injector.PacketFilterManager.handlePacket(PacketFilterManager.java:590) [ProtocolLib-4.5.1.jar:4.5.1]
at com.comphenix.protocol.injector.PacketFilterManager.invokePacketRecieving(PacketFilterManager.java:557) [ProtocolLib-4.5.1.jar:4.5.1]
at com.comphenix.protocol.injector.netty.ProtocolInjector.packetReceived(ProtocolInjector.java:350) [ProtocolLib-4.5.1.jar:4.5.1]
at com.comphenix.protocol.injector.netty.ProtocolInjector.onPacketReceiving(ProtocolInjector.java:315) [ProtocolLib-4.5.1.jar:4.5.1]
at com.comphenix.protocol.injector.netty.ChannelInjector.decode(ChannelInjector.java:538) [ProtocolLib-4.5.1.jar:4.5.1]
at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:249) [server.jar:1_8_R3]
at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:149) [server.jar:1_8_R3]
at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:333) [server.jar:1_8_R3]
at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:319) [server.jar:1_8_R3]
at com.comphenix.protocol.injector.netty.ChannelInjector$2.channelRead(ChannelInjector.java:259) [ProtocolLib-4.5.1.jar:4.5.1]
at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:333) [server.jar:1_8_R3]
at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:319) [server.jar:1_8_R3]
at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:163) [server.jar:1_8_R3]
at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:333) [server.jar:1_8_R3]
at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:319) [server.jar:1_8_R3]
at net.minecraft.server.v1_8_R3.LegacyPingHandler.channelRead(LegacyPingHandler.java:88) [server.jar:1_8_R3]
at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:333) [server.jar:1_8_R3]
at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:319) [server.jar:1_8_R3]
at io.netty.handler.timeout.ReadTimeoutHandler.channelRead(ReadTimeoutHandler.java:150) [server.jar:1_8_R3]
at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:333) [server.jar:1_8_R3]
at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:319) [server.jar:1_8_R3]
at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:787) [server.jar:1_8_R3]
at io.netty.channel.epoll.EpollSocketChannel$EpollSocketUnsafe.epollInReady(EpollSocketChannel.java:722) [server.jar:1_8_R3]
at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:326) [server.jar:1_8_R3]
at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:264) [server.jar:1_8_R3]
at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:116) [server.jar:1_8_R3]
at java.lang.Thread.run(Thread.java:748) [?:1.8.0_312]
Caused by: java.lang.ClassNotFoundException: com.comphenix.protocol.injector.temporary.TemporaryPlayerFactory
at java.net.URLClassLoader.findClass(URLClassLoader.java:387) ~[?:1.8.0_312]
at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:105) ~[server.jar:1_8_R3]
at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:90) ~[server.jar:1_8_R3]
at java.lang.ClassLoader.loadClass(ClassLoader.java:418) ~[?:1.8.0_312]
at java.lang.ClassLoader.loadClass(ClassLoader.java:351) ~[?:1.8.0_312]
... 32 more
> [14:02:59 ERROR]: Parameters:
net.minecraft.server.v1_8_R3.PacketHandshakingInSetProtocol@222025b7[
a=47
hostname=no, thank u
port=25565
d=LOGIN```

### CODE REVIEW

File: bungeeguard-spigot/src/main/java/me/lucko/bungeeguard/spigot/listener/ProtocolHandshakeListener.java [hunk 1]
```diff
@@ -63,8 +65,16 @@ public void registerAdapter(Plugin plugin) {
...
+
         Adapter(Plugin plugin) {
             super(plugin, ListenerPriority.LOWEST, PacketType.Handshake.Client.SET_PROTOCOL);
+            try {
+                Class.forName("com.comphenix.protocol.injector.temporary.MinimalInjector");
+                plugin.getLogger().info("Using newer ProtocolLib support.");
+            } catch (ClassNotFoundException e) {
+                plugin.getLogger().info("Using legacy ProtocolLib support.");
+                isLegacyProtocolLib = true;
+            }
```
Comment (stijnb1234) [inline bungeeguard-spigot/src/main/java/me/lucko/bungeeguard/spigot/listener/ProtocolHandshakeListener.java:77]: Maybe replace newer/legacy by v4 and v5?

**Review by stijnb1234 — APPROVED** [review]
LGTM, small point of discussion.

**Comment by stijnb1234** [comment]
Closes #88, #92, #96

**Comment by lucko** [comment]
Thanks :)

**Pull request merged**